### PR TITLE
Port the `theories` folder

### PR DESCRIPTION
**Pull request by riccardobrasca**

### CODE REVIEW

File: data/menus.yaml [hunk 1]
```diff
@@ -38,6 +38,14 @@
     - Undergraduate maths: undergrad.html
     - Wiedijk's 100 theorems: 100.html
 
+- title: Theories
+  items:
+    - Category Theory: theories/category_theory.html
+    - Linear Algebra: theories/linear_algebra.html
+    - Natural Numbers: theories/naturals.html
+    - Sets and finite sets: theories/sets.html
+    - Topology: theories/topology.html
```
Comment (jcommelin) [inline data/menus.yaml:47]: Not sure what the best ordering is, but certainly I would put natural numbers at the top and categories at the bottom.
```suggestion
    - Natural Numbers: theories/naturals.html
    - Linear Algebra: theories/linear_algebra.html
    - Sets and finite sets: theories/sets.html
    - Topology: theories/topology.html
    - Category Theory: theories/category_theory.html
```

File: templates/theories/category_theory.md [hunk 2]
```diff
@@ -58,7 +39,7 @@ We use `≅` for isomorphisms.
 ### Functors
 
 We use `⥤` (`\func`) to denote functors, as in `C ⥤ D` for the type of functors from `C` to `D`.
-(Unfortunately `⇒` is reserved in [`logic.relator`](https://github.com/leanprover-community/mathlib/blob/master/src/logic/relator.lean), so we can't use that here.)
+(Unfortunately `⇒` is reserved in [`Mathlib.Logic.Relator`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Logic/Relator.html), so we can't use that here.)
```
Comment (PatrickMassot) [inline templates/theories/category_theory.md:42]: This is no longer true! This stuff is now in Mathlib and we could easily scope the exotic relator notation.

Comment (riccardobrasca) [reply templates/theories/category_theory.md]: Fixed, thanks.

**Comment by fpvandoorn** [comment]
I didn't check if all the information is up-to-date, but it's probably better to merge this with a few outdated bits than to have nothing.

**Pull request merged**